src/Core/Services/Implementations/UserService.cs

```diff
@@ -1226,18 +1226,29 @@ public async Task<bool> CanAccessPremium(ITwoFactorProvidersUser user)
             {
                 return false;
             }
-            if (user.GetPremium())
+
+            return user.GetPremium() || await this.HasPremiumFromOrganization(user);
+        }
+
+        public async Task<bool> HasPremiumFromOrganization(ITwoFactorProvidersUser user)
+        {
+            var userId = user.GetUserId();
+            if (!userId.HasValue)
             {
-                return true;
+                return false;
             }
-            var orgs = await _currentContext.OrganizationMembershipAsync(_organizationUserRepository, userId.Value);
-            if (!orgs.Any())
+
+            // orgUsers in the Invited status are not associated with a userId yet, so this will get
+            // orgUsers in Accepted and Confirmed states only
+            var orgUsers = await _organizationUserRepository.GetManyByUserAsync(userId.Value);
+
+            if (!orgUsers.Any())
             {
                 return false;
             }
             var orgAbilities = await _applicationCacheService.GetOrganizationAbilitiesAsync();
-            return orgs.Any(o => orgAbilities.ContainsKey(o.Id) &&
-                orgAbilities[o.Id].UsersGetPremium && orgAbilities[o.Id].Enabled);
+            return orgUsers.Any(ou => orgAbilities.ContainsKey(ou.OrganizationId) &&
+                orgAbilities[ou.OrganizationId].UsersGetPremium && orgAbilities[ou.OrganizationId].Enabled);
```

Comment:
IMO, this can be improved by writing it as:
```c#
return orgUsers.Any(ou => orgAbilities.TryGetValue(ou.OrganizationId, out var orgAbility) 
                          && orgAbility.UsersGetPremium && orgAbility.Enabled);
```

Reply:
I agree, I've done this and formatted it better